### PR TITLE
added flaky_code and success_delay_ms metadata for testing against grpc

### PR DESCRIPTION
**Pull request by bizob2828**
## Proposed Release Notes
 * Adds flaky_code and success_delay_ms handling of flaky grpc connections to infinite tracing
 
## Links
Closes: #599 

## Details
 * This is intended for non-production use and mostly for the Node.js agent team
 * Manually testing the 2 new metadata options

Simulate `flaky_code`
```
NEWRELIC_GRPCCONNECTION_METADATA_FLAKY=95 NEWRELIC_GRPCCONNECTION_METADATA_FLAKY_CODE=7 node --r newrelic server.js
```

produces:
```
{"v":0,"level":40,"name":"newrelic","hostname":"C02F81D3MD6P","pid":43995,"time":"2021-04-28T19:26:10.762Z","msg":"Span stream error. Code: [7]: Chaos flaky disconnect","component":"grpc_connection"}
{"v":0,"level":10,"name":"newrelic","hostname":"C02F81D3MD6P","pid":43995,"time":"2021-04-28T19:26:10.762Z","msg":"gRPC Status Received [7]: Chaos flaky disconnect","component":"grpc_connection"}
```

Simulate `success_delay_ms`:
```
```

produces:

### CODE REVIEW

File: lib/grpc/connection.js [hunk 1]
```diff
@@ -163,22 +169,27 @@ class GrpcConnection extends EventEmitter {
...
     return metadata
   }
 
+  /**
+   * Adds test metadata used to simulate connectivity issues
+   * when appropriate env vars are set
+   *
+   * @param {Metadata} metadata
+   */
+  _getTestMetadata(metadata, env) {
```
Comment (michaelgoin) [inline lib/grpc/connection.js:183]: minor but would prob rename to `_setTestMetadata` given this is mutating the `metadata` collection VS just returning new items.

**Review by michaelgoin — APPROVED** [review]
Looks good. One minor naming comment you can decide if you want to incorporate or not.

File: lib/grpc/connection.js (hunk 1 above)
Comment (bizob2828) [reply lib/grpc/connection.js:183]: good call

**Review by michaelgoin — APPROVED** [review]
👍

**Pull request merged**